VideoLocker/src/main/java/org/edx/mobile/user/Account.java

```diff
@@ -0,0 +1,201 @@
+package org.edx.mobile.user;
+
+import android.support.annotation.NonNull;
+import android.support.annotation.Nullable;
+
+import com.google.gson.annotations.SerializedName;
+
+import java.util.ArrayList;
+import java.util.Date;
+import java.util.List;
+
+public class Account {
+    @SerializedName("username")
+    @NonNull
```

Comment:
We should ideally strive to make the app crash-proof even in the case of
API bugs. You're right that defaults would be overridden if the fields are
explicitly set to null in the JSON response though, but it's more resilient
than not using defaults at all. Best would be to have JSON ignore null
fields during deserialization, but there doesn't seem to be a simple
setting to achieve that, short of using a custom TypeAdapter.

On Fri, Sep 25, 2015 at 8:14 PM, Brian Guertin <notifications@github.com>
wrote:

> In VideoLocker/src/main/java/org/edx/mobile/user/Account.java
> <https://github.com/edx/edx-app-android/pull/372#discussion_r40440365>:
>
> > @@ -0,0 +1,201 @@
> > +package org.edx.mobile.user;
> > +
> > +import android.support.annotation.NonNull;
> > +import android.support.annotation.Nullable;
> > +
> > +import com.google.gson.annotations.SerializedName;
> > +
> > +import java.util.ArrayList;
> > +import java.util.Date;
> > +import java.util.List;
> > +
> > +public class Account {
> > +    @SerializedName("username")
> > +    @NonNull
>
> I added annotations to match the API documentation. If they are wrong,
> then there is an API bug that should be fixed.
>
> Even if we initialized with non-null values, the API could make them null
> again by return null for that field, so that wouldn't be fool-proof.
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/edx/edx-app-android/pull/372/files#r40440365>.
>